engine/notifier_test.go

```diff
@@ -107,11 +107,15 @@ func TestNotifier_ManyConsumers(t *testing.T) {
 		time.Sleep(10 * time.Microsecond)
 	}
 
+	nPendingWorkers := pendingWorkers.Load()
 	// require that all workers got a notification
 	require.Eventuallyf(t,
-		func() bool { return pendingWorkers.Load() == 0 },
+		func() bool {
+			nPendingWorkers = pendingWorkers.Load()
+			return nPendingWorkers == 0
+		},
 		3*time.Second, 100*time.Millisecond,
-		"still awaiting %d workers to get notification", pendingWorkers.Load(),
+		"still awaiting %d workers to get notification", nPendingWorkers,
```

Comment:
Perhaps you still misunderstand my point about the error comments from earlier? This is exactly the opposite of what we want :/

Reply:
Note, at Line 114, we update the pendingWorks every time we check the condition, if the condition in the end still didn't meet, then the cached `nPendingWorks` is the most recent, and we will read `nPendingWorks` at that moment.

require.Eventually takes in the variable not the value, it won't read the variable until waited for timeout.